### PR TITLE
Extended double factorials for complex numbers

### PR DESCRIPTION
**Pull request by czgdp1807**
I have extended the factorial2 method in factorials.py
for complex functions by using the concept of gamma functions

### CODE REVIEW

**Comment by jksuom** [comment]
There is no natural extension of `factorial2` to the complex domain. Please consider [this comment](https://github.com/sympy/sympy/pull/10362#issuecomment-170344298).

File: sympy/functions/combinatorial/factorials.py [hunk 1]
```diff
@@ -359,6 +361,12 @@ def eval(cls, arg):
         # TODO: extend this to complex numbers?
 
         if arg.is_Number:
+            if arg.is_complex:
```
Comment (asmeurer) [inline sympy/functions/combinatorial/factorials.py]: is_complex applies to every finite number, including real numbers.

**Comment by normalhuman** [comment]
@Abdullahjavednesar This isn't "SymPy's turn" situation.  Incorrect logic (complex numbers will not pass `is_Number` check), no tests, no response to jksuom's comment...

**Comment by czgdp1807** [comment]
This PR can be closed since #10362 has already worked on this and merged.
Closing. 
I was a beginner at that time, apologies for the delay.

**Pull request closed without merging**